File: bindings/go/binding_test.go
package tree_sitter_whim_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_whim "github.com/carthage-software/tree-sitter-whim/bindings/go"
)

func TestCanLoadGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_whim.Language())
	parser := tree_sitter.NewParser()
	defer parser.Close()
	if err := parser.SetLanguage(language); err != nil {
		t.Fatalf("Error loading Whim grammar: %v", err)
	}

	tree := parser.Parse([]byte("final class Box<T> {}\n$box = new Box::<string>();"), nil)
	defer tree.Close()
	if tree.RootNode().HasError() {
		t.Error("Error parsing Whim syntax")
	}
}

File: bindings/node/binding_test.js
import assert from "node:assert";
import { test } from "node:test";
import Parser from "tree-sitter";
import language from "./index.js";

test("can load grammar", () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(language));
});

test("can parse Whim syntax", () => {
  const parser = new Parser();
  parser.setLanguage(language);
  const tree = parser.parse(`
    final class Box<T> {
      public function __construct(public T $value) {}
    }

    $box = new Box::<string>('whim');
  `);

  assert.equal(tree.rootNode.hasError, false);
  assert.equal(tree.rootNode.namedChild(0).type, "class_declaration");
  assert.equal(tree.rootNode.namedChild(1).type, "expression_statement");
});
